warehouse/query-core/src/main/java/datawave/query/tracking/ActiveQuery.java

```diff
@@ -0,0 +1,151 @@
+    }
+    
+    synchronized public void beginCall(Range range, CallType type) {
+        Map<Range,Long> rangeMap = this.startTimeMap.get(type);
+        if (rangeMap == null) {
+            rangeMap = new HashMap<>();
+            this.startTimeMap.put(type, rangeMap);
+        }
+        rangeMap.put(range, System.currentTimeMillis());
+        activeRanges.add(range);
+        setInCall(type, true);
+    }
+    
+    synchronized public void endCall(Range range, CallType type) {
+        Long val = getNumCalls(type);
+        this.numCallsMap.put(type, (val.longValue() + 1));
+        
+        Map<Range,Long> rangeMap = this.startTimeMap.get(type);
+        if (rangeMap != null) {
+            Long start = rangeMap.get(range);
+            if (start != null) {
+                getTimer(type).update(System.currentTimeMillis() - start, TimeUnit.MILLISECONDS);
+                rangeMap.remove(range);
+                setInCall(type, false);
+            }
+        }
+    }
+    
+    synchronized public void recordStats(Document document, QuerySpan querySpan) {
+        this.documentSizeBytes = document.sizeInBytes();
+        this.lastSourceCount = querySpan.getSourceCount();
+        this.lastSeekCount = querySpan.getSeekCount();
+        this.lastNextCount = querySpan.getNextCount();
+    }
+    
+    synchronized public int removeRange(Range range) {
+        this.activeRanges.remove(range);
+        return this.activeRanges.size();
+    }
+    
+    @Override
+    synchronized public String toString() {
+        return snapshot().toString();
+    }
+    
+    synchronized private long getNumCalls(CallType type) {
+        Long val = this.numCallsMap.get(type);
+        if (val == null) {
+            val = 0l;
+            this.numCallsMap.put(type, val);
+        }
+        return val;
+    }
+    
+    synchronized private long totalElapsedTime() {
+        return System.currentTimeMillis() - this.initTs;
+    }
+    
+    synchronized private boolean isInCall() {
+        int numInCallTotal = 0;
+        for (Integer numInCall : this.inCallMap.values()) {
+            numInCallTotal += numInCall;
+        }
+        return numInCallTotal > 0;
+    }
+    
+    synchronized private Timer getTimer(CallType type) {
+        Timer t = timerMap.get(type);
+        if (t == null) {
+            t = new Timer(new SlidingWindowReservoir(this.windowSize));
+            this.timerMap.put(type, t);
+        }
+        return t;
+    }
+    
+    synchronized private void setInCall(CallType type, boolean inCall) {
+        Integer numCalls = this.inCallMap.get(type);
+        if (numCalls == null) {
+            numCalls = new Integer(0);
+        }
+        int newNumCalls;
+        if (inCall) {
+            newNumCalls = numCalls.intValue() + 1;
+        } else {
+            newNumCalls = numCalls.intValue() - 1;
+        }
+        if (newNumCalls < 0) {
+            newNumCalls = 0;
```

Comment:
Perhaps this requires a warning message as it implies we are ending more calls than we are starting.....

Reply:
changed from info to warn.